java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/TopicPartitionOffsetProvider.java

```diff
@@ -0,0 +1,113 @@
+package io.lenses.java.streamreactor.connect.azure.eventhubs.source;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+
+/**
+ * This class represents an abstraction over OffsetStorageReader that can be freely called by Azure
+ * EventHub Kafka Consumers when it was initialized once. It helps consumers to find out which
+ * offset were already committed inside Kafka Connect.
+ */
+@Slf4j
+public final class TopicPartitionOffsetProvider {
```

Comment:
Why is there a need to have a static instance? 
Can getOffset method not be a static method which also receives OffsetStorageReader as parameter?

Reply:
My logic was: this component is Connector-wide which means it's a good candidate for app-wide Singleton.

We can make it non-singleton but then big downside will be that I will initialize it in Connector class (because it has `OffsetStorageReader`) and pass it to every class (whether it uses it or not) so it'll eventually be passed to KafkaConsumers. IMO it's not worth it, but feel free to suggest another approach.

Reply:
Each Task runs in its isolated class loader. So the singleton approach does not apply. I will take it offline to explain.

Reply:
OH!!! Now I get why in [this comment](https://github.com/lensesio/stream-reactor/pull/1033#discussion_r1506151573) you say that the static won't work. I was initially under the impression that Every Task (and Connector) share the same classloader. I'll have amend it shortly, but offline explanation will definitely be appreciated!

Reply:
Okay, I moved initialization to the Task, this should be way more streamlined now, thanks!